project1/client/input_parser/input_parser.h

```diff
@@ -0,0 +1,105 @@
+#ifndef PROJECT1_INPUT_PARSER_H
+#define PROJECT1_INPUT_PARSER_H
+
+#include "../../wire_protocol/wire_protocol.h"
+#include "../../server/file_handler/my_file_handler.h"
+#include "ftp_messages.pb.h"
+#include <sstream>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace client::input_parser {
+
+    /**
+     * @brief Assists in mapping input commands for FTP client to corresponding
+     * protobuf requests.
+     *
+     * @note There's definitely some redundancy here. Will likely change
+     * this class to have templated CreateReq functions in project 2.
+     */
+    class InputParser {
+    public:
+        enum ReqType {GETF, PUTF, DEL, LS, CD, MKDIR, PWD, QUIT};
+
+        /**
+         * @brief Initializes relevant info about this input command
+         * @param cmd the user input
+         * @note assumes correct syntax
+         */
+        InputParser(std::string& cmd);
+        ~InputParser();
+
+        /**
+         * @brief tells user the request type
+         * @return the request type. user relies on this information to determine
+         * which CreateReq function to call
+         */
+        ReqType GetReqType();
+
+        /**
+         * @brief getter for filename
+         * @return the filename
+         */
+        std::string GetFilename();
+        /**
+         * creates a get request
+         * @return the request
+         */
+        ftp_messages::GetRequest CreateGetReq();
```

Comment:
One minor thing you can do is take advantage of overloading like I did in my `Agent` class. That way, you don't have to explicitly remember to call the right version of this function.